yb-voyager/src/tgtdb/conn_pool.go

```diff
@@ -40,15 +40,17 @@ type ConnectionParams struct {
 
 type ConnectionPool struct {
 	sync.Mutex
-	params       *ConnectionParams
-	conns        chan *pgx.Conn
-	nextUriIndex int
+	params                    *ConnectionParams
+	conns                     chan *pgx.Conn
+	connIdToPreparedStmtCache map[uint32]map[string]bool // cache list of prepared statements per connection
+	nextUriIndex              int
 }
 
 func NewConnectionPool(params *ConnectionParams) *ConnectionPool {
 	pool := &ConnectionPool{
-		params: params,
-		conns:  make(chan *pgx.Conn, params.NumConnections),
+		params:                    params,
+		conns:                     make(chan *pgx.Conn, params.NumConnections),
+		connIdToPreparedStmtCache: make(map[uint32]map[string]bool, params.NumConnections),
```

Comment:
When a connection is closed don't we drop items from this map?

Reply:
yeah that should be done. thanks